.github/actions/nf-test/action.yml

```diff
@@ -0,0 +1,115 @@
+name: "nf-test Action"
+description: "Runs nf-test with common setup steps"
+inputs:
+  profile:
+    description: "Profile to use"
+    required: true
+  shard:
+    description: "Shard number for this CI job"
+    required: true
+  total_shards:
+    description: "Total number of test shards(NOT the total number of matrix jobs)"
+    required: true
+  paths:
+    description: "Test paths"
+    required: true
+
+runs:
+  using: "composite"
+  steps:
+    - name: Setup Nextflow
+      uses: nf-core/setup-nextflow@v2
+      with:
+        version: "${{ env.NXF_VERSION }}"
+
+    - name: Set up Python
+      uses: actions/setup-python@42375524e23c412d93fb67b49958b491fce71c38 # v5
```

Comment:
I am curious why pin to a specific hash, versus major version.  I see we do that [elsewhere](https://github.com/nf-core/fastquorum/blob/1f6d25fc9c0b30f2c756fd844bdaa7c352a82e07/.github/workflows/linting.yml#L17C32-L17C72), which I presume gets updated when the nf-core template gets updated, but how do we keep this one updated?  Probably a question for another day.

Reply:
I don't know the answer to that, best asking on the nf-core slack.

Reply:
I guess the answer might be explained by [this recent compromised action](https://semgrep.dev/blog/2025/popular-github-action-tj-actionschanged-files-is-compromised/), with the advice in that article being: 
> Generally, pin all GitHub Actions to specific commit SHAs (rather than version tags) you know are safe. In this case, it appears that all versions are compromised.